### ISSUE TITLE
WriteConfigAs for toml config type results in error

### ISSUE THREAD

**Issue by Viv1k** [spf13/viper#488]
Following error is returned while saving config as toml

`While marshaling config: cannot convert type []interface {} to Tree`

**Comment by ovaltzer** [comment]
I have just encountered the same issue.
after digging up, we found out that when writing toml, the data is assumed to be without any structs, but only primitives. if you will write this file as json, load it and write to toml, it will magically work.
but it's not a solution.. 
basically in this line:
https://github.com/spf13/viper/blob/b5e8006cbee93ec955a89ab31e0e3ce3204f3736/viper.go#L1673		`deepestMap[lastKey] = value`
value object can be interface as there is no process which makes this map\slice\etc...

**Comment by Viv1k** [comment]
Yes, you are correct. Even, I ended up doing the same... :smiley: 

I had one more question (separate from this topic)  
I read config into a structure and then modified that structure. Is it possible to marshall and write new structure into a config file using viper? 
I don't want to iterate over the complete structure and write into a new viper object.

**Comment by ovaltzer** [comment]
i'm trying to use https://github.com/fatih/structs
https://github.com/fatih/structs/blob/ebf56d35bba727c68ac77f56f2fcf90b181851aa/structs.go#L81-L85

**Comment by Javirln** [comment]
Hello, is it planned to fix the issue?

**Comment by ViRb3** [comment]
Bump, same issue here, forcing me to migrate to a different config format.
